src/MapboxStyleParser.ts

```diff
@@ -281,18 +304,31 @@ export class MapboxStyleParser implements StyleParser<Omit<MbStyle, 'sources'>>
    * @param spriteName Name of the sprite
    * @return the url that returns the single image
    */
-  getIconImage(spriteName: string): (string|undefined) {
+  async getIconImage(spriteName: string): Promise<Sprite | undefined> {
     if (!spriteName) {
       return;
     }
     if (!this.spriteBaseUrl) {
       return;
     }
-    // TODO update endpoint as soon as api specification was made
-    let url: string = '/sprites/?';
-    url += 'name=' + spriteName;
-    url += '&baseurl=' + encodeURIComponent(this.spriteBaseUrl);
-    return url;
+
+    const response = await fetch(this.spriteBaseUrl + '.json');
```

Comment:
should fetching be the default behavior? I'd say yes, but would it be sensible to provide a way around?
I can imagine there are scenarios where you would not want to fetch, for example if the parser is not run in a browser.

Then a sprite declaration would only make sense if you could provide the necessary meta data via another way, maybe to the constructor?

Reply:
https://github.com/geostyler/geostyler-mapbox-parser/issues/310